### ISSUE TITLE
Resource view creation errors in new flask view

### ISSUE THREAD

**Issue by wardi** [ckan/ckan#4475]
### CKAN Version if known (or site URL)
master

### Please describe the expected behaviour
creating resource views should work and be covered by at least the most basic unit tests

### Please describe the actual behaviour
creating resource views doesn't work, see:
https://github.com/ckan/ckan/pull/4062/files#r219926627
https://github.com/ckan/ckan/pull/4062/files#r219927711

hacking around those leads to a "View type: Missing value" form error because the view type is no longer passed as part of the URL as it was in the pylons controller.

### What steps can be taken to reproduce the issue? 
Click "New View" button on a resource

**Comment by wardi** [comment]
for reference here are the hacks that get me to the third failure
```diff
diff --git a/ckan/views/resource.py b/ckan/views/resource.py
index c88bcb4eb..58b713478 100644
--- a/ckan/views/resource.py
+++ b/ckan/views/resource.py
@@ -612,18 +612,13 @@ class EditResourceViewView(MethodView):
     def post(self, package_type, id, resource_id, view_id=None):
         context, extra_vars = self._prepare(id, resource_id)
 
-        to_preview = request.POST.pop(u'preview', False)
+        data = clean_dict(dict_fns.unflatten(tuplize_dict(parse_params(
+            request.form, ignore_keys=CACHE_PARAMETERS))))
+        data.pop(u'save', None)
+        to_preview = data.pop(u'preview', False)
         if to_preview:
             context[u'preview'] = True
-        to_delete = request.POST.pop(u'delete', None)
-        data = clean_dict(
-            dict_fns.unflatten(
-                tuplize_dict(
-                    parse_params(request.form, ignore_keys=CACHE_PARAMETERS)
-                )
-            )
-        )
-        data.pop(u'save', None)
+        to_delete = data.pop(u'delete', None)
         data[u'resource_id'] = resource_id
 
         try:
@@ -680,6 +675,7 @@ class EditResourceViewView(MethodView):
                 return base.abort(404, _(u'View not found'))
 
         view_type = view_type or request.args.get(u'view_type')
+        data = {}
         data[u'view_type'] = view_type
         view_plugin = lib_datapreview.get_view_plugin(view_type)
         if not view_plugin:

```

**Comment by smotornyuk** [comment]
Fixed by #4476
